src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs

```diff
@@ -245,7 +275,8 @@ public async Task InitializeAsync(IEnumerable<FunctionMetadata> functions, Cance
 
             var workerConfig = _workerConfigs.Where(c => c.Description.Language.Equals(_workerRuntime, StringComparison.InvariantCultureIgnoreCase)).FirstOrDefault();
 
-            if (workerConfig == null && (functions == null || functions.Count() == 0))
+            // We are skipping this check for multi-language environments because they use multiple workers and thus doesn't honor 'FUNCTIONS_WORKER_RUNTIME'
+            if (workerConfig == null && (functions == null || functions.Count() == 0) && !_environment.IsMultiLanguageRuntimeEnvironment())
```

Comment:
Having a cache in static variable was causing issues while running tests in `CloudTest`. Removed the caching because these checks are at host start and string operations might not be very expensive given the size of `APP_KIND` is not very huge. 
What do you think?

![Screenshot 2022-07-13 054830](https://user-images.githubusercontent.com/104004368/178622440-e783b626-b176-4588-8589-1c72b00aff98.jpg)

Reply:
One thing which can be done maybe is that, cache it twice at [ScriptHost.cs](https://github.com/Azure/azure-functions-host/pull/8449/files#diff-6d3165a3c204458bede6be5209ce39874c572862494969cb364f31e0d1c7d1b3)[ScriptHost.cs](https://github.com/Azure/azure-functions-host/pull/8449/files#diff-6d3165a3c204458bede6be5209ce39874c572862494969cb364f31e0d1c7d1b3)  and [RpcFunctionInvocationDispatcher.cs](https://github.com/Azure/azure-functions-host/pull/8449/files/cc66cc10e644fd57d867dd0c59004faf908abd95#diff-5a241b8ff64ccf9edb3fc0cc63256d99705ba2ffa011019fa8371d16d86b6cd6) level. Should we cache it twice?

Reply:
Issue is resolved now!